rustls/src/client/builder.rs

```diff
@@ -14,23 +14,23 @@ use super::client_conn::Resumption;
 use pki_types::{CertificateDer, PrivateKeyDer};
 
 use alloc::sync::Arc;
-use core::marker::PhantomData;
 
-impl<C: CryptoProvider> ConfigBuilder<ClientConfig<C>, WantsVerifier> {
+impl ConfigBuilder<ClientConfig, WantsVerifier> {
     #[cfg(feature = "ring")]
     /// Choose how to verify server certificates.
     pub fn with_root_certificates(
         self,
         root_store: impl Into<Arc<webpki::RootCertStore>>,
-    ) -> ConfigBuilder<ClientConfig<C>, WantsClientCert> {
+    ) -> ConfigBuilder<ClientConfig, WantsClientCert> {
         ConfigBuilder {
             state: WantsClientCert {
+                provider: self.state.provider,
                 cipher_suites: self.state.cipher_suites,
                 kx_groups: self.state.kx_groups,
                 versions: self.state.versions,
                 verifier: Arc::new(webpki::WebPkiServerVerifier::new(root_store)),
             },
-            side: PhantomData,
+            side: core::marker::PhantomData,
```

Comment:
i will put this back, but i did it for (what I think is) a good reason: 

```rust
#[cfg(any(feature = "dangerous_configuration", feature = "ring"))]
use core::marker::PhantomData;
```

this feels pretty gross; i can't really express how little i like having to cfg-gate imports to avoid `unused_imports` noise.

Reply:
Seems reasonable to compromise on this if it avoids having to gate the import. Maybe worth a note about that in the `CONTRIBUTING.md`?